### ISSUE TITLE
Recommend using nullglob in loops

### ISSUE THREAD

**Issue by hseg** [koalaman/shellcheck#2029]
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
PATH=~/.local/bin:"$PATH"
for i in ~/.local/opt/*/bin; do PATH="$i:$PATH"; done
export PATH

```

#### Here's what shellcheck currently says:

No guidance

#### Here's what I wanted or expected to see:

This will iterate once, even if the glob fails to match. Use `shopt -s nullglob` on bash, or explicitly test for file existence on POSIX shell.

(in fact, the default behaviour for globs is problematic in general, but I don't know enough to say whether a general nullglob advice is appropriate)

**Comment by hseg** [comment]
Judging by <https://unix.stackexchange.com/a/204944>, the appropriate advice seems to be `nullglob` for loops and `failglob` for nonloops.

**Comment by hseg** [comment]
Indeed, this is the behaviour implemented by [fish](https://fishshell.com/docs/2.4/index.html#expand)
